pmtiles/server.go

```diff
@@ -320,6 +336,7 @@ func (server *Server) getTile(ctx context.Context, httpHeaders map[string]string
 	}
 	return status, headers, data
 }
+
 func (server *Server) getTileAttempt(ctx context.Context, httpHeaders map[string]string, name string, z uint8, x uint32, y uint32, ext string, purgeEtag string) (int, map[string]string, []byte, string) {
 	rootReq := request{key: cacheKey{name: name, offset: 0, length: 0}, value: make(chan cachedValue, 1), purgeEtag: purgeEtag}
```

Comment:
I thought about trying to short-circuit the request up here by comparing the request etag to the root etag to return a 306/412 without having to fetch the tile, but that doesn't work because we need to go out to the actual tile to check if the etag on the overall file has changed

Reply:
Yeah this introduces some tricky situations. Up to now our focus on ETags has been to always return some correct result, and never return a "broken" result like reading stale ranges from a new file. 

It seems like to present a semantically correct view over the bucket we must invoke at least one (and usually just one) storage request for every request into the server, so that we are checking the etag liveness every time. 

Originally I thought the "one etag for all resources" would be better because of that shortcut but it feels wrong to possibly return stale conditional requests when the underlying blob has actually changed. 

I'm learning towards making unique etags the default and not having any option.

Reply:
If the only downside to generating unique ETags is computational overhead, is there an alternative to md5 that is (even) faster? anything that doesn't require cgo is fair game. I don't think we need to promise etags are any specific checksum or even consistent across implementations (JS/workers/go-pmtiles)

Reply:
Switched to https://github.com/cespare/xxhash - 1.25 million tile requests from tilesiege using archive etag:

```
user    2m17.600s
sys     1m14.435s
```

and with unique per-tile etags:

```
user    2m22.171s
sys     1m16.552s
```

Since the performance difference is negligible I just made it the default with no way to use archive etag.

Reply:
I wonder what raw fnv1a 64-bit (like is used in `pmtiles convert`) results are like instead of xxhash? if that is faster than md5 and close to xxhash that would remove a dependency

Reply:
fnv1a:

```
user    3m22.259s
sys     1m11.702s
```

Seems like it's a big enough improvement to warrant the extra dep. Also it was already a transient dependency, this just moves it to a direct dependency. The executable size diff is ~50kb on mac build.